src/libostree/ostree-repo-commit.c

```diff
@@ -408,18 +453,176 @@ add_size_index_to_metadata (OstreeRepo        *self,
             g_hash_table_lookup (self->object_sizes, e_checksum);
           _ostree_write_varuint64 (buffer, e_size->archived);
           _ostree_write_varuint64 (buffer, e_size->unpacked);
+          g_string_append_c (buffer, (gchar) e_size->objtype);
```

Comment:
The core incompatible change being this.

Reply:
Indeed. However, as I stated in a couple other comments, I really doubt people are using it. Primarily because there's no exported way to read the variable length integers. You'd have to duplicate the code from ostree as you can see in this old xdg-app commit that removes its usage - https://github.com/alexlarsson/xdg-app/commit/be71ceb31f715074d10d4c4e37e39ba72f0c7549.

I did a github code search on `ostree.sizes`, which unfortunately is sorta useless because it ignores the `.`. However, I trawled through [many pages of it](https://github.com/search?q=ostree.sizes&type=Code) and the only actual references were copies of ostree itself (including ours which expands on upstream) and xdg-app (as noted above).

Furthermore, because of the variable length integers, no code could have been asserting a fixed length. What could be done is asserting that there were no more bytes in the array after reading the 2nd integer. I didn't see that anywhere, though.

So, I think it is understandable to want to use a different key. In fact, that's how I was first approaching this. However, after reviewing the facts above, I came to the conclusion that it was highly unlikely that this would break anything besides the ostree testsuite.

Reply:
OK yep, let's get this in then!